src/zc/lockfile/__init__.py

```diff
@@ -83,15 +90,24 @@ def __init__(self, path):
             _lock_file(fp)
         except:
             fp.seek(1)
-            pid = fp.read().strip()[:20]
+            content = fp.read().strip()
             fp.close()
-            if not pid:
-                pid = 'UNKNOWN'
-            logger.exception("Error locking file %s; pid=%s", path, pid)
+            if content_template == '{pid}':
+                # Original exception message format when using the default
+                # lock file template
+                pid = content[:20] if content else 'UNKNOWN'
+                logger.exception("Error locking file %s; pid=%s", path, pid)
+            else:
+                # Include the first 40 characters of lock file contents for
+                # custom lock file templates
+                logger.exception('Error locking file %s; content: "%s%s"',
+                                 path, content[:40],
+                                 '...' if len(content) > 40 else '')
             raise
 
         self._fp = fp
-        fp.write(" %s\n" % os.getpid())
+        fp.write(" %s\n" % content_template.format(pid=os.getpid(),
+                                                   hostname=LazyHostName()))
```

Comment:
That may've been premature!

I now see that on Windows the code uses byte-range locking to lock just the 1st byte.  Perhaps locking on Windows prevents other processes from *reading* too?  If that's the case, then my suggestion was seriously wrong!